database/cache.go

```
@@ -209,6 +237,33 @@ func unsafeAddUserToCache(u UserT) {
 	cache.userSlice = append(cache.userSlice, u)
 }
 
+// unsafe functions aren't concurrency safe
+func unsafeAddVoteToCache(u int32, q int32) {
+	if u < 1 {
+		// u must be greater than zero to be a valid UserID
+		return
```

this shall not fail silently